### DISCUSSION TITLE
compress images

### DISCUSSION THREAD

**Discussion by AhmedHashem2104** [adonisjs/core#3377] [Help]
public async createProduct({ response , request , auth }: HttpContextContract){

    const payload = await request.validate(ProductValidator)

    const images = await request.files('imgs' , {
      size: '2mb',
      extnames: ['jpg', 'png'],
    
    })

    if(images.length === 0){

      return response.status(422).json({
        message: `No images attached`
      })

    }
    else{

    for (const image of images) {
      await image.move(Application.publicPath('uploads') , {
        name: `${uuidv4()}.${image.extname}`,
        overwrite: true, // overwrite in case of conflict
      })
    }

    var newImages = Array()

    for(let i = 0 ; i < images.length ; i++){
      newImages[i] = images[i].fileName
    }

  }

  return images
 }

how can i compress images in this code

**Comment by RomainLanz** [comment]
Hey @AhmedHashem2104! 👋🏻 

By using a compression library, feel free to use whatever you want. I personally use [sharp](https://www.npmjs.com/package/sharp).

There is also a community package called [adonis-responsive-attachment](https://www.npmjs.com/package/adonis-responsive-attachment).

**Reply by AhmedHashem2104** [reply]
can you show me example of using sharp in adonisjs

**Reply by AhmedHashem2104** [reply]
like in my code

**Reply by AhmedHashem2104** [reply]
i want to compress jpg or png images

**Reply by feniljariwala82** [reply]
Read the documentation! It's very easy to implement
